[Project] Refunds, returns, discounts daily metric updates (#147)
fivetran/dbt_shopify: CHANGELOG.md

# dbt_shopify v1.6.0

[PR #145](https://github.com/fivetran/dbt_shopify/pull/145) includes the following updates:

## Schema/Data Changes
**18 total changes • 9 possible breaking changes**

| Data Model(s) | Change type | Old | New | Notes |
| ------------- | ----------- | --- | --- | ----- |
| [shopify__refund_lines](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__refund_lines) or<br>[shopify_gql__refund_lines](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__refund_lines) | New model | - | - | New final model at the refund line item grain. Exposes per-line refund financials, product context, and restock classification. |
| [shopify__refunds](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__refunds) or<br>[shopify_gql__refunds](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__refunds) | New model | - | - | New final model at the refund transaction grain. Aggregates line item financials and includes discrepancy adjustments for accurate reconciliation. |

> | [shopify__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__daily_shop) or<br>[shopify_gql__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__daily_shop) | Data change | `refund_subtotal`, `refund_total_tax`, and `count_orders_with_refunds` bucketed by order creation date | Now bucketed by refund creation date | Aligns with Shopify Finance report behavior, which attributes refunds to the date the refund was issued. |

Reviewer: "Bucketed" is a bit confusing. At the core, the issue was that refunds were being recorded per the order date, we've since updated it to be applied per the refund date. Suggestion to adjust wording to match this.
```suggestion
| [shopify__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__daily_shop) or<br>[shopify_gql__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__daily_shop) | Data change | `refund_subtotal`, `refund_total_tax`, and `count_orders_with_refunds` applied per order creation date | Refund fields now applied per refund creation date | Aligns with Shopify finance report behavior, which attributes refunds to the date the refund was issued. |
```

| [stg_shopify_gql__return_line_item](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__return_line_item) | New model | - | - | New GraphQL staging model for return line item records. |
| [stg_shopify_gql__return_shipping_fee](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__return_shipping_fee) | New model | - | - | New GraphQL staging model for return shipping fee records. |
| [stg_shopify_gql__order_line_refund](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__order_line_refund) | New columns | - | `price_pres_amount`, `price_pres_currency_code`, `price_shop_amount`, `price_shop_currency_code`, `restocked` | Unit price at refund time (shop and presentment) and Shopify-native restock boolean. |
| [stg_shopify_gql__refund](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__refund) | New column | - | `updated_at` | Timestamp of the last update to the refund record. | 

## Under the Hood

> - Introduces the `shopify_gql_using_return` variable (default `false`) to opt into return modeling. When enabled, `shopify_gql__refund_lines` is enriched with return lifecycle and return line item detail columns. `shopify_gql__refunds` is enriched with return context and return shipping fee columns.

Reviewer: Is this really an under the hood change?

| [shopify__refunds](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__refunds) or<br>[shopify_gql__refunds](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__refunds) | New model |  |  | New final model at the refund transaction grain. Aggregates line item financials and includes discrepancy adjustments for accurate reconciliation. |
| [shopify__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__daily_shop) or<br>[shopify_gql__daily_shop](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__daily_shop) | Data change | `refund_subtotal`, `refund_total_tax`, and `count_orders_with_refunds` bucketed by order creation date | Now bucketed by refund creation date | **Possible breaking change** New refund metrics now likely to change values. Aligns with Shopify Finance report behavior, which attributes refunds to the date the refund was issued. |

> | [shopify__customers](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__customers)<br> and [shopify__customer_emails](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__customer_emails) or<br>[shopify_gql__customers](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__customers)<br> and [shopify_gql__customer_emails](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__customer_emails) | Data change | `lifetime_total_discount` and `avg_discount_per_order` sourced from `order_line.total_discount` | Now sourced from `discount_allocation` | **Possible breaking change** New discount metrics now likely to change values. `discount_allocation` is the more reliable source for order line discount values. |

Reviewer 2: ```suggestion
| [shopify__customers](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__customers)<br> and [shopify__customer_emails](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__customer_emails) or<br>[shopify_gql__customers](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__customers)<br> and [shopify_gql__customer_emails](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__customer_emails) | Data change | `lifetime_total_discount` and `avg_discount_per_order` sourced from `order_line.total_discount` | Now sourced from `discount_allocation` | **Possible breaking change:** New discount metrics now likely to change values. `discount_allocation` is the more reliable source for order line discount values. |
```

| [shopify__orders](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify__orders) or<br>[shopify_gql__orders](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.shopify_gql__orders) | New fields |  | `gross_sales`, `discounts`, `returns`, `net_sales` | Finance metrics at the order level, matching the REST `shopify__orders` columns. |
| [stg_shopify_gql__order_line_refund](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__order_line_refund) | New columns |  | `price_pres_amount`, `price_pres_currency_code`, `price_shop_amount`, `price_shop_currency_code`, `is_restocked` | **Possible breaking change** Users with `order_line_refund_pass_through_columns` configured may encounter duplicate column errors if any passthrough columns overlap with the new columns.  |
| [stg_shopify_gql__refund](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__refund) | New column |  | `updated_at` | Timestamp of the last update to the refund record. |

> | [stg_shopify_gql__return](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__return), [stg_shopify_gql__return_line_item](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__return_line_item), [stg_shopify_gql__return_shipping_fee](https://fivetran.github.io/dbt_shopify/#!/model/model.shopify.stg_shopify_gql__return_shipping_fee) | New models |  |  | Opt-in GraphQL staging models for return data. Can be enabled/disabled using `shopify_gql_using_return` variable. When enabled, `shopify_gql__refund_lines` is enriched with return lifecycle and return line item detail columns, and `shopify_gql__refunds` is enriched with return context and return shipping fee columns. |

Author: Added.
